app/src/main/java/ru/practicum/android/diploma/presentation/viewmodels/SearchViewModel.kt

```diff
@@ -0,0 +1,135 @@
+package ru.practicum.android.diploma.presentation.viewmodels
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import ru.practicum.android.diploma.domain.SearchInteractor
+import ru.practicum.android.diploma.domain.models.Vacancy
+import ru.practicum.android.diploma.ui.state.SearchScreenState
+import ru.practicum.android.diploma.util.Options
+import ru.practicum.android.diploma.util.ResponseData
+import ru.practicum.android.diploma.util.debounce
+
+class SearchViewModel(
+    private val searchInteractor: SearchInteractor
+) : ViewModel() {
+    companion object {
+        const val ITEMS_PER_PAGE = 20
+
+        const val ONE_SECOND = 1000L
+        const val TWO_SECONDS = 2000L
+    }
+
+    private val screenStateLiveData = MutableLiveData<SearchScreenState>(SearchScreenState.Default)
+    private var currentPage = 0
+    private var maxPages = 1
+    private var isNextPageLoading = false
+    private var mainRequest = ""
+    private var requestNextPage = ""
+    private val _vacancyIsClickable = MutableLiveData(true)
+    var vacancyIsClickable: LiveData<Boolean> = _vacancyIsClickable
```

Comment:
```suggestion
    val vacancyIsClickable: LiveData<Boolean> = _vacancyIsClickable
```
лайвдату только немьютабельной можно отдавать на вью, это касается и ссылки